### DISCUSSION TITLE
phase out IdGeneratorStrategyInterpreter

### DISCUSSION THREAD

**Discussion by gavinking** [hibernate/hibernate-orm#5914] [Design Proposals]
I would like to propose that we essentially drop `IdGeneratorStrategyInterpreter` and the current logic for instantiating an id generator for a JPA `@GeneratedValue`, and:

- just plain hardwire-in the interpretation of the `GenerationType`s in terms of our build-in `Generator`s, and
- start instantiating them via constructors which accept annotations, as we do with our new stuff.

(A motivating factor here is to migrate to the use of `UuidGenerator`, which is currently impossible.)

Of course we would need to keep a map of named id generator -> defining annotation instance (a `TableGenerator`, `SequenceGenerator`, or `GenericGenerator` annotation).

This might be a more-radical proposal than it needs to be, so perhaps @sebersole can come up with a "compromise"?

**Comment by gavinking** [comment]
> Of course we would need to keep a map of named id generator -> defining annotation instance

Excuse me, that's not quite right. We would have a map of name to id generator *instance*. We would instantiate the generator when we encountered the `TableGenerator`, `SequenceGenerator`, or `GenericGenerator` annotation.

**Comment by sebersole** [comment]
https://hibernate.atlassian.net/browse/HHH-14496

**Reply by gavinking** [reply]
OK, great.

**Comment by sebersole** [comment]
We should close this one imo to avoid split discussion

**Reply by gavinking** [reply]
OK sure
